### PR TITLE
Add support for Device Trust Mtls redirect remediation

### PR DESCRIPTION
**Pull request by santoshmale-okta**
## Description:

Add support for Device Trust Mtls redirect remediation

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Reviewers:
@haishengwu-okta @pradeepdewda-okta @okta/device-platform @manishagarwal-okta 

### Issue:

- [OKTA-373640](https://oktainc.atlassian.net/browse/OKTA-373640)

### CODE REVIEW

File: src/v2/ion/responseTransformer.js [hunk 1]
```diff
@@ -169,6 +169,25 @@ const convertRedirectIdPToSuccessRedirectIffOneIdp = (settings, result, lastResu
...
+const convertRedirectToSuccessRedirectForMtls = (result) => {
+  if (Array.isArray(result.remediations)) {
+    const redirectRemediations = result.remediations.filter(r => r.name === RemediationForms.REDIRECT_DEVICE_TRUST);
+    if (redirectRemediations.length === 1 && result.remediations.length === 1) {
+      const successRedirect = {
+        name: RemediationForms.SUCCESS_REDIRECT,
+        href: redirectRemediations[0].href,
+        value: [],
+      };
+      result.remediations = [ successRedirect ];
```
Comment (jane-kuang-okta) [inline src/v2/ion/responseTransformer.js:187]: nit: Don't need extra spaces between brackets and variable. `[successRedirect]`

File: src/v2/ion/responseTransformer.js [hunk 2]
```diff
@@ -230,8 +249,9 @@ const convert = (settings, idx = {}, lastResult = null) => {
 
   injectIdPConfigButtonToRemediation(settings, result);
   if (!result.messages) {
-    // Only redirect to the IdP if we are not in an error flow
+    // Only redirect to the IdP or Device Trust Mtls if we are not in an error flow
```
Comment (jane-kuang-okta) [inline src/v2/ion/responseTransformer.js:252]: Is it possible that there is both a `redirect-device-trust` and `redirect-idp` remediation form? If so would that cause any problems? Or is it fine that `convertRedirectToSuccessRedirectForMtls` would just override result.remediations with the Mtls href?

**Review by jane-kuang-okta — COMMENTED** [review]
Small question and nit!

File: src/v2/ion/responseTransformer.js (hunk 2 above)
Comment (santoshmale-okta) [reply src/v2/ion/responseTransformer.js:252]: It should never happen. IDP routing happens at the Identify Step. Device Trust MTLS redirect happens post identify step after the user is identified. Also, I added the check `result.remediations.length === 1` to make sure to redirect to DT one only when it is the only remediation in the list

File: playground/mocks/data/idp/idx/device-trust-mtls-redirect.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  "stateHandle": "inRUXNhsc6Evt7GAb8DPAA",
+  "version": "1.0.0",
+  "expiresAt": "2020-04-13T22:55:53.000Z",
+  "intent": "LOGIN",
+  "remediation": {
+    "type": "array",
+    "value": [
+      {
+        "name": "redirect-device-trust",
+        "href": "http://dv1-devtrust.okta1.com/user/settings/certificate/challenge?relaystate=abcdUXb8DPNhsc6Evt7GXYa123",
```
Comment (manishagarwal-okta) [inline playground/mocks/data/idp/idx/device-trust-mtls-redirect.json:11]: nit relaystate should be relayState + orgSubDomain is also needed.

File: src/v2/ion/responseTransformer.js [hunk 4]
```diff
@@ -169,6 +169,25 @@ const convertRedirectIdPToSuccessRedirectIffOneIdp = (settings, result, lastResu
   }
 };
 
+/**
+ * IFF there is one `redirect-device-trust` remediation form, widget will automatically redirect to
```
Comment (manishagarwal-okta) [inline src/v2/ion/responseTransformer.js:173]: IFF or If?

**Review by manishagarwal-okta — APPROVED** [review]

File: src/v2/ion/responseTransformer.js (hunk 2 above)
Comment (jane-kuang-okta) [reply src/v2/ion/responseTransformer.js:252]: Accidentally glossed over the `result.remediations.length === 1`, makes sense!

**Review by jane-kuang-okta — APPROVED** [review]

File: playground/mocks/data/idp/idx/device-trust-mtls-redirect.json (hunk 3 above)
Comment (santoshmale-okta) [reply playground/mocks/data/idp/idx/device-trust-mtls-redirect.json:11]: not really. We do not pass org subdomain here, thats a diff endpoint.

**Comment by santoshmale-okta** [comment]
Side channeled with @haishengwu-okta . He had a suggestion to just re-use the `success-redirect` remediation since the DT remediation is identical to it. So, this PR is no longer needed. Made the change as part of this PR https://github.com/okta/okta-core/pull/51820. Closing this one.

**Pull request closed without merging**